<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
  <head>
    <title>Beyond Short Read Sequencing</title>
    <meta charset="utf-8" />
    <meta name="author" content="Alexis Lucattini" />
    <link rel="stylesheet" href="libs\remark-css-0.0.1\Long_Reads_QLD.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# Beyond Short Read Sequencing
## Applications of Long Reads
### Alexis Lucattini
### Mon 10 Jun 2019

---


## Beyond Short Reads

.pull-left[
.large[Benefits]
* Long-range haplotyping.
* Methylation with single-molecule level sequencing.
* Scaffolding and genome assembly
* Resolving of large structural variations
]

.pull-right[
.large[Platforms]
* ONT &lt;img src=./images/ont_logo.png width=20%&gt;
* PacBio &lt;img src=./images/pacbio_logo.png width=20%&gt;
]


???

Notes:
What are the benefits of long reads.
With short read sequencing, haplotyping information can be conveyed as long as there's no homogeneity for 300 bps. This often means that a SNP on one exon cannot be determined to be linked to a SNP on another exon for a given individual.
Both ONT and PacBio offer single-molecule sequencing, the DNA is not amplified prior to entering the flowcell meaning additional modifications such as methylation can be detected with high coverage.
More commonly long-reads are known for resolving scaffolds for genome assembly. Short-reads simply cannot overcome repeat regions longer than their read-length.

# TODO 
Add ONT and Pacbio image on the columns tag

# ALL TODOS
PACBIO TOOLS -
HOW DOES THE PACBIO WORK?
GENOME ASSEMBLIES
STRUCUTURAL VARIATION TOOLS
ONT AND PACBIO DATASETS
VAGRANT VM

---

class: long-slide

## Bioinformatics is a bottleneck
* Novel file types 
  + multi-fast5s
  + unaligned BAMs

* Many ways to do one thing, benchmarking needed.

* Getting help and resources, community still in development.

* Denovo assemblies have always been really hard.

???

Class: ALL

Notes:

Bioinformatics seems to be a bit of an afterthought.
Raw data off each of the machines comes in a different format, meaning bioinformatics algorithms often need to be built from the ground up.
Denovo assemblies will remain really hard as labs try to complete tasks in the most cost-efficient way.

---

class: long-slide

## New methods for standard tasks
.pull-left[
.baby-bear[
**Basecalling**
* Compressed Signal Data
* Local alignments

**Data QC**
* Read length, yields, read quality
* Fastq *indicitive* of read quality.
]
]
.pull-right[
.baby-bear[
**Adapter trimming**
* Nanopore reads have adapters too.
* PacBio reads can be one sequence repeated many times.

**Alignment**
* Different parameter settings required.
]
]

???

Class: ONT, PacBio

Notes: 

While the standard fastq format is still used, getting to fastqs is not trivial. For ONT, this involves running digital signal data through an RNN to convert into sequencing values. QC values are no longer using interop files or Q30 scores, but we want to look at read length, the yield of the data and the sequencing quality, which can be much more varied when using long-read sequencing platforms.

---

class: long-slide

# Minimap2

* Current gold-standard aligner for ONT and PacBio genomic data.
* Make sure the index is an 'ont-index' or 'pacbio-index' when generating an index file.
* Alignments come with CS-string.

.pull-left[
[GitHub](https://github.com/lh3/minimap2)
]
.pull-right[
[GitHub Guide](https://github.com/lh3/minimap2#users-guide)
]

???

CAN BE BOth PROMETHION AND PACBIO

---

class: long-slide

# The CS tag and MD tag
.pull-left[
.baby-bear[
* CS tag - verbose version of MD tag
  + Easier for parsing.
  + Relatively new.
  + MD tag still required for some parsers.
    + Pipe through `samtools calmd` prior to sorting.
]
]
.pull-right[
.baby-bear[
* CS tag example:
  + `:6-ata:10+gtc:4*at:3`
  + `:[0-9]+ is an identical block`
  + '-[ACTG]+ is a deletion`
  + '+[ACTG]+ is an insersion
  + '*XY' is a substitution with base X subsituted for base y.
]
]

???

Minimap2 is the standard aligner for ONT and PacBio genomic data.
It's extremely fast and has some useful output.

MAYBE TODO Add in slide with minimap2 and nglmr, or put this in the SV slide.

---

class: ont-slide

## Nanopore Sequencing - PMI

.pull-left[
**Technical**
* Very long sequencing reads
* Single-molecule
* Real-time
* Portable
* Low accuracy
* Lower yields
]

.pull-right[
**Financial**
* Minimal start-up cost
* Cheap for small uses of data
* Relatively expensive per base
]

# TODO

Place ONT VIDEO in here

---

class: ont-slide

## Basecalling methods
### GPU &amp; CPU.
* GPUs much faster. .small[Guppy 1.8.3 is the current basecalling version]

* Different algorithms give slightly different results.

* .small[Check out Ryan Wick's [base-calling comparison repo](https://github.com/rrwick/Basecalling-comparison)]


???

Not just the algorithm, but the training data it a has been exposed to. We often see that the quality of human data to be a bit higher than other data sets.

---

class: ont-slide

## WUB
* Software package repository from ONT.
* Similar to pysam API.
* Handy plotting tools.

.pull-left[
[GitHub](https://www.github.com/nanoporetech/wub)
]
.pull-right[
[Docs](https://wub.readthedocs.io/en/latest/)
]

???

The wub package contains a large range of plotting tools,  
for nanopore sequencing. 

Some of the plots aren't perfect but everything comes out in a picklable.
These can be easily repimported into python and be re-plotted.

---

class: ont-slide

## WUB cont.

.pull-left[
&lt;img src=./images/basestats.png width=100%&gt;
]
.pull-right[
&lt;img src=./images/per_read_accuracies.png width=100%&gt;

]

???

We've seen that the mapping rates 


---

class: ont-slide

## NanoPlot

.pull-left[
.baby-bear[
* Plot run metrics for a MinION run
* [Repo on GitHub](https://github.com/wdecoster/NanoPlot)
* Also see 
  + NanoComp
  + NanoStat
  + NanoFilt
  + NanoLyse
]
]

.pull-right[
&lt;img src="https://github.com/wdecoster/NanoPlot/raw/master/examples/scaled_MappingQualityvsAverageBaseQuality_kde.png" width="100%"&gt;
]

???



---

class: ont-slide

## Porechop, Filtlong, Unicycler

Ryan Wick's repos set for:
* [Demultiplexing / Trimming Adapters](https://github.com/rrwick/porechop)
* [Removing poor quality reads](https://github.com/rrwick/filtlong)
* [Generating Circular-hybrid assemblies](https://github.com/rrwick/unicycler)

???

NANOPORE SPECIFIC

---

class: ont-slide

# Nanopolish

Jared T. Simpson's [nanopolish](https://github.com/jts/nanopolish) uses signal-level (fast5) data to:
* Detecting methylation
* Calling SNP level variants
* Polishing genomes through consensus

???

NANOPORE SPECIFIC

---

class: ont-slide

# Tombo

**ONT's methylation caller.**

* Provides output of regions of significance
* Can be captured as a 'wiggle' file to be viewed in a genome browser.

.pull-left[
[GitHub](https://github.com/nanoporetech/tombo)
]
.pull-right[
[Docs](https://nanoporetech.github.io/tombo/tutorials.html)
]

???

NANOPORE SPECIFIC

---

class: ont-slide

## Haplotyping-methylation combo

.center[
&lt;img src="images/scott_paper_methylation_differences.png" width="55%"&gt;
]

.pull-left[
[BioRxiv Paper](https://www.biorxiv.org/content/early/2018/10/17/445924)
]
.pull-right[
[Scott's GitHub Haplotype-Methylome Repo](https://github.com/scottgigante/haplotyped-methylome)
]

???

NANOPORE SPECIFIC

---

class: ont-slide

## AGRF PromethION in focus

Kit-9, sample sensitive, great N50!

.pull-left[
&lt;img src=./images/example.yield.png width=80%&gt;
]
.pull-right[
&lt;img src=./images/example.weighted.hist.png width=80%&gt;
]

---

class: ont-slide

## Some tutorial references

[Tim Kahlke - Toast 2018](https://github.com/timkahlke/toast2018)  
VM based tutorial. Covers QC to methylation to assembly.


[Hadrien Gourle](https://github.com/HadrienG/tutorials/blob/master/docs/nanopore.md)  
Simple fastq to assembly notebook.

???

These linkes are a little old but the data application is still relevant. File input is now close to multi-fast5s

---

class: pacbio-slide

# Pacific Biosciences

[&lt;img src=./images/pacbio_fridge.jpg width=80%&gt;](https://www.youtube.com/watch?v=WMZmG00uhwU&amp;feature=youtu.be)

???

Greater maturity in bioinformatics components. Much of what needs to be done can be done via the smrtlink.
Read number restricted by the number of wells - rather than the amount of sequencing. Data comes out as a really large movie that is then converted to unaligned bam files.
Provide estimates of sequencing length and yield from slides. 

---

# PacBio - PMI

.pull-left[
**Technical**
* Single-molecule
* Sequence-by-replication approach
* Long-reads with high consensus.
* Well-supported bioinformatics
* Finite limit of number-of-molecules.
* Output:
  + 10-25 Gbs Amplicon
  + 5-10 Gbs gDNA - n50 10-15Kb
]

.pull-right[
**Financial**
* Large-start up cost.
* Relatively expensive per base
]

???

The pacbio currently provides a comparable level of throughput to a MinION run but has benefits of having an improved error-rate and a reduction in systematic bias.

---

# Introduction to SMRTLINK

Bioinformatics at a click-of-a-button!
* Easy stand-alone workflows 
  + cover much of the capability of the device.
  + runs scatter-gather workflows across an HPC
* BAM to CSS.
* SV Calling
* De novo Assembly.
* SNP and INDEL calling on long-range amplicons.

???

Our SMRT-LINK runs on a singularity container, an is an on-demand device.
So only running when we need it!
SMRTLINK and its associated command-line version offer scatter-gather algorithms for each of the different services that we offer. I plan to put a link up on a mini-tutorial of how to set up your own SMRTLINK portal.

---

# Subread vs CCS Read 

&lt;img src='image/subread_to_ccs.png', width=100%&gt;

???

The subread contains the smrtbell adapters, converting re-sequenced reads into a concensus single read. 
This increases the accuracy in the read with every additional cycle. Often with genomic DNA, only one cycle is completed. There exists a trade-off between accuracy and length.

Lima is a PacBio developed algorithm for converting subreads into concensus reads.

---

# PacBio IsoSeq

.pull-left[
+ Full length transcripts.
+ High-consensus rates.
+ Well defined bioinformatics workflows.
+ Complex normalisation methods - pre-QC.
]
.pull-right[
&lt;img src='images/exone_in_genes.png' width=80%&gt;
]

???

Utilisation of CCS, full-length transcriptome. Goes through many times allowing for full assembly. Provide a picture of an entire gene for this slide.
Find short read comparison. Exon proportions slide to place in.

---

# SV Calling with PacBio

.pull-left[
* PacBio supported pipeline is pbsv
  + available in smrtlink.
* Other community-developed pipelines include:
  + [sniffles](https://github.com/fritzsedlazeck/Sniffles
)
  + [smrtsv2](https://github.com/EichlerLab/smrtsv2
)
]
.pull-right[
* Both also compatible with ONT data.
]

???

Easy work with SMRTLINK but can use community pipelines as well.
Get a picture of ngmlr vs minimap2.

https://www.pacb.com/products-and-services/analytical-software/whole-genome-sequencing/
https://github.com/fritzsedlazeck/Sniffles


---

# Targeted Sequencing with PacBio

???

# TODO:
LAA amplicons can resolve haplotyping information and assembly of small important regions. Reducing the cost of sequencing yield whilst also increasing coverage of the important region.

---

# Denovo Assembly

???

#TODO 
Talk about Flye, wtdg2 and then hgap4.
What can be done with the overall layout and consensus, copy from the previous work at nanopore dunedin.
Add in the links completed by Ben Langmead.
Picture of bandage in here. Length L law torsten seeman.

---

# Final Remarks.

???

Not a one-size fits all approach.
Yield vs accuracy trade-off in per-flowcell

---

# Future steps

## ONT - R10 pore, greater accuracy, targeted approach in development


## PacBio - Greater yield, more wells per flowcell.


???

Like the rivalry between Microsoft and Apple, two different operating systems with different pros and cons have focused hard on overcoming their cons, essentially resulting in two products which are now close to identical.

---

# Other things that may interest you

Working with Singularity.
Working with Vagrant.
SMRTLINK On Demand - email me and I'll give you a brief on how to set it up.


???

SMRTLINK On Demand. - coming soon.
Singularity.
Vagrant.

---

# Thank you slide

???

#TODO
Add thank yous
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "pygments",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();</script>

<script>
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
