<!DOCTYPE html>
<html>
  <head>
    <title>Fantastic Hints</title>
    <meta charset="utf-8">
    <meta name="author" content="Alexis Lucattini" />
    <link rel="stylesheet" href="libs/remark-css-0.0.1/Getting_Started_with_a_MinION.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# Fantastic Hints
## and where to find them
### Alexis Lucattini
### Fri 09 Feb 2018

---


## Chemistry history 
.left-column[.center[R7 to R9.4]
&lt;img src=images/yield_plots_old.png width=100%&gt;
.center[.tiny[Image Credit Scott Gigante]]
]
.pull-right[.center[R9.4 April-2017]  
&lt;img src=images/yield_plots_latest.png width=110%&gt;
]


???

When we first started using a MinION, a good run was around 400 Mb of data.
We celebrated in the 1GB club using the R9 chemistry in June 2016 and then found 10 GB in April last year.
Our R9.5 runs were actually a bit lower than the R9.4. This has been a common finding of regular nanopore users.
Recently announced are R9.4.1 and R9.5.1 which we are yet to run.

---

# Quality Control

* Quantification.
    + Qubit
* Purity.
    + Nanodrop
* Distribution.
    + TapeStation

???

We take a three pronged approach to library preparation.
We want to make sure that the sample mass is sufficient to start the prep.
It's important that our sample is purely DNA or RNA with minimal salt concentration.
And that the sample is of the distribution that we expect.

---

.my-header[&lt;h1 style="display:inline;top:100px;" &gt;More is more:&lt;/h1&gt; &lt;h3 style="display:inline;"&gt;when using HMW&lt;/h3&gt;]

A 1 `\(\mathrm{\mu}\)`g starting input is standard protocol...  

$$\text{if } \frac{1\ \mathrm{\mu g}}{8\ \mathrm{kb}} \approx 0.2\ \mathrm{pmol} $$

$$\therefore \frac{50\ \mathrm{kb}}{0.2\ \mathrm{pmol}} \approx 6\ \mathrm{\mu g} $$

???

Reactions throughout the library prep and sequencing are based on the number of available ends.
A HMW library of 50 kb requries an input of 6 ?g to achieve equal molarity of a 8kb library of 1?g  

---

# 'Mix by pipetting'

* DON'T!  
   + .small[Or use wide-bore pipettes only.]&lt;br&gt;&lt;br&gt;  
* 70% - 80% Ethanol.  &lt;br&gt;&lt;br&gt;
* Lower ratio on bead-clean ups  
   + .small[0.4X for HMW DNA]

???

Great care should be taken throughout the library prep.
Mixing using regular pipette tips can damage a library.
Increasing stringency on bead cleanups can reduce the number of short fragements in a library.
When a high starting input is used with HMW, you will break more, so you need to clean more. 

---

# Some further tips

.pull-left[

Storing flowcells
   * .small[Avoid dynamic fridges]
   * .small[Insulate from cycles]
]

.pull-right[

Bubbles are bad.
   &lt;img src=http://www.science-practice.com/assets/york-minion/flowcell.jpg
    width="120%"&gt;
]

???

A frozen flowcell is a bad flowcell. 
Flowcells should be kept at the same frigerated temperature. 
Fridges that cycle will reduce the lifespan of these flowcells.
The back of the fridge can sometimes freeze. Avoid placing flowcells there.
Flowcells should be layed completely flat, to prevent bubbles and prevent further damage from bubbles.

Any region of the flowcell that has been exposed to air is destroyed.
In your early days of priming you will probably do this unintentionally.
Practise priming with a used flowcell.


---

## Protocols are just guidelines...

&lt;a href="https://docs.google.com/document/d/1iBIiQIPCZQRZ13lBuv7MJL9OIs1KPooomqGkQcoywwg/edit"&gt;One-Pot ligation&lt;/a&gt; from Josh Quick (UK)

&lt;a href="https://www.protocols.io/protocols?key=nanopore"&gt;&lt;img src="https://synbiobeta.com/wp-content/uploads/sites/4/2016/10/protocols.io_.jpg"&gt;&lt;/a&gt;

---


class: center

# The nanopore pipeline.

&lt;img src=images/pipeline.png width="110%" float="left"&gt;

???

Prior to running any analysis on the data coming from the MinION,
we need to basecall the data. That is converting the analogue signal here.


https://twitter.com/biomickwatson/status/647395496114450432
https://www.drive5.com/usearch/manual/fastq_fig.jpg

---


class: h1-with-images

.center[&lt;h1&gt;What The Fast5? &lt;/h1&gt; 
&lt;img margin-top=0px src=images/fast5_file.png width="65%"&gt;]

???

It is possible to run a nanopore sequencing device mulitple types without ever actually seeing the inside of a fast5.
The .fast5 format is based on the hdf5 format from https://support.hdfgroup.org/HDF5/
For what a .bcl file is to Illumina, a .fast5 file is to nanopore. 
Other than the squiggle data as shown above, there are a few other metrics in the fast5 file that may be of interest.


---

.my-header[&lt;h1 style="display:inline;top:100px;" &gt;Basecalling:&lt;/h1&gt; &lt;h3 style="display:inline;"&gt;From .fast5 to .fastq.&lt;/h3&gt;]

All basecalling, is computationally intensive &lt;a href="https://github.com/alexiswl/poreduck"&gt;.small[(use a HPC)]&lt;/a&gt;.

Head to &lt;a href="https://github.com/rrwick/Basecalling-comparison"&gt;Ryan Wick's Github&lt;/a&gt; for detailed information on aligners.

&lt;table class="table" style="margin-left: auto; margin-right: auto;"&gt;
 &lt;thead&gt;
  &lt;tr&gt;
   &lt;th style="text-align:left;"&gt; BaseCallers &lt;/th&gt;
   &lt;th style="text-align:left;"&gt; Optimised &lt;/th&gt;
   &lt;th style="text-align:left;"&gt; Bonus &lt;/th&gt;
  &lt;/tr&gt;
 &lt;/thead&gt;
&lt;tbody&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; Albacore &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; CPU &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Common &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; Guppy &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; GPU &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Fast &lt;/td&gt;
  &lt;/tr&gt;
&lt;/tbody&gt;
&lt;/table&gt;


???

You have a plethora of options, albacore, guppy and Metrichor.
For high-quality yield, straight to the HPC.
How do I place data on the HPC?
Either https://github.com/alexiswl/poreduck
Or Nick Lomans helpful hints: https://github.com/nickloman/nanopore-basecalling-scripts

---

# Alignment algorithms

The alignment parameters for nanopore reads are different to that of other sequencing machines.

* Minimap2 

* bwa-mem

* GraphMap

???

Alignment algorithms for nanopore are different to those from standard Illumina reads. 
Illumina reads are more accurate, but shorter, requires a different approach.



---

.center[&lt;h1&gt; The Elephant in the Room &lt;/h1&gt;
&lt;img src=https://pbs.twimg.com/media/CClEkuLW0AAlzFE.jpg width=80%&gt;
]

???

We got our PromethION on the 21st of July last year.
It has been a slow progress but we have started to see some results.

---

.center[&lt;h1&gt;LAMBDA : LSK108 : Feb 18&lt;/h1&gt;
&lt;img src=images/yield_promethion_20180202.png width=45%&gt;
]

???

This is our yield from our lambda run last Friday.
As we hadn't basecalled at this point, we weren't aware that the yield was continuing to rise. ONT contacted us and said, would you mind restarting the run for another 16 hours. We'd like to see if we can hit 50 Gb.

---

class: genomics-innovationh-hub, gih-slide-number

&lt;img src=images/gih_partnerships.png width=110%&gt;

---

class: genomics-innovationh-hub, gih-slide-number

# Hub Goals

* Building innovation ecosystems.
* Stepping up to the innovation challenge.
* Providing access to state-of-the-art technologies
* Managing cost efficiency

.pull-right[
Contact: &lt;a href="mailto:azadeh.seidi@agrf.org.au"&gt;Azadeh Saedi&lt;/a&gt;
.kinda-small[Website: &lt;a href="https://genomicsinnovationhub.org.au"&gt;GenomicsInnovationHub.org.au&lt;/a&gt;]]


???

* Building innovation ecosystems: 
    + through common research, collaboration, education and technology development.
* Stepping up to the innovation challenge: 
    + bringing resources together to diffuse cutting-edge genomic technologies.
* Providing access to state-of-the-art technologies: 
    + access to Australiaâ€™s first 10X Genomics Chromium system, as well as early access to cutting-edge technologies such as the Oxford Nanopore Technologies PromethION.
* Managing cost efficiency: 
    + open access to a portfolio of cutting-edge technologies and competencies.
---

.pull-left[
* R&amp;D:
    + Azadeh Saedi
    + Ken McGrath
    + Kirby Siemering
    + Lavinia Gordon
* Lab Ops:
    + Matthew Tinning
* IT:
    + Chris Hunt
    + Douglas Morrison
    + Gismon Thomas
* PR:
    + Desley Pitcher
]

.pull-right[
* WEHI
    + Matthew Ritchie
    + Quentin Gouil
    + Scott Gigante
    + Chris Woodruff
]
    </textarea>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "pygments",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {window.dispatchEvent(new Event('resize'));});
(function() {var d = document, s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler"); if (!r) return; s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }"; d.head.appendChild(s);})();</script>

<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre']
  }
});
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
