<!DOCTYPE html>
<html>
  <head>
    <title></title>
    <meta charset="utf-8">
    <link rel="stylesheet" href="libs\remark-css-0.0.1\SingleCellSummary.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">


# Single Cell Bioinformatics: Where to start

The number of packages is forever growing.  

Some good guides to package sets are at:

* [scrna-tools.org](scrna-tools.org)  
* [awesome single cell GitHub repo](https://github.com/seandavi/awesome-single-cell)  

Today we'll go through three.

+ Monocle  

+ Seurat  

+ ScPipe  

+ SIMLR  

---

# Monocle

* Clustering and counting cells  

* Constructing single-cell trajtories  

* Differential expression between cell-types  

* Differential expression between time-points of a single cell type  

* Pseudotime tracking  

[Repo](http://cole-trapnell-lab.github.io/monocle-release/docs/#differential-expression-analysis)

???

Monocle was one of the first single-cell packages around.
It has a bulk of citations and provides a comprehensive list of tools.

Straight forward installation process, pulls UMI transcript count matrices out of cell ranger.
Monocle will also normalise the data.

Psuedotime - how far through a cell-fate has an individual cell gone through?
Appropriate use-case would be tracking the cell-fates throughout a developmental process.

---

## Monocole plots - Gene expression in pseudotime  
.center[
&lt;img src=http://cole-trapnell-lab.github.io/monocle-release/images/vignette/plot_diff_res_pt-1.png width="70%"&gt;
]
---

## Monocole plots - Gene expression between cell-types

&lt;img src=http://cole-trapnell-lab.github.io/monocle-release/images/vignette/plot_diff_res_multi-1.png width="120%"&gt;

---

# Seurat

* Discovery and exploration of cell types along with cell-filtering

* Generate QC metrics

* Easily imports 10X data.

* Can pass data into Monocle.

???

QC metrics include showing the number of UMI per cell barcode in violin plot.
The percentage of mitochondrial DNA (an indicator of dead/dying cells)
The number of genes per cell also in a violin plot.

Useful to assign cell-type to identify clusters as a secondary confirmation to Monocle.

---

## Seurat pics

&lt;img src=images/seurat_cluster.png width="120%"&gt;


---

# ScPipe

.pull-left[
* Testing for cell-outliers.

* Another good QC metrics to use.
]
.pull-right[
&lt;img src=images/sc_pipe_plot.png width="120%"&gt;
]
---

# SIMLR

* Visualisation  

* Dimension reduction.  

* De novo discovery of cell-types  

* Beyond requirements for just 10X data - research focus.
    </textarea>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "pygments",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function() {
  var d = document, s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})();</script>

<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre']
  }
});
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
