<!DOCTYPE html>
<html>
  <head>
    <title>Longranger, Lariat and Loupe</title>
    <meta charset="utf-8">
    <meta name="author" content="Alexis Lucattini" />
    <link rel="stylesheet" href="libs\remark-css-0.0.1\linked_reads.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# Longranger, Lariat and Loupe
## The ins and outs of the 10X pipeline
### Alexis Lucattini
### Thu 19 Jul 2018

---


### Another visual introduction to linked-reads:

.small[Because you can never watch this video too many times]
.center[&lt;a href="https://10xgenomics.wistia.com/medias/2r0nal3n3j"&gt;&lt;img src=./images/intro_video_thumbnail.png width=68%&gt;]

???

Add linked reads video here.
https://10xgenomics.wistia.com/medias/2r0nal3n3j

70s in.

---

# The Longranger pipeline.

.center[&lt;img src=./images/10X_DIAGRAM.png width=50%&gt;]

???

An important note is that the bcl2fastq stage is separate from the standard illumina bcl2fastq.
The longranger mkfastq is essentially a wrapper aroudn bcl2fastq with a couple of tinkerings.

---

# 10x mkfastq
.pull-left[
.baby-bear[
* Wrapper of bcl2fastq
* Input
  + Requires input and output directories
  + A samplesheet
  + A base-mask parameter.
* Output
  + Fastq files split by 10X kit ID .tiny[(SI-GA-01)]
    + Barcode/Index QC Values.
    + GEM estimates
]]
.pull-right[
*  `*_L1.fastq.gz` contains the i7 index
*  `*_R1.fastq.gz` contains the fastq information. &lt;br&gt;
&lt;img src=./images/index_orientation.png width=130%&gt;
]

???

---

# Longranger basic
.pull-left[
Input: Fastq
Approx run time: A few hours.
Output:
*  unaligned Bam or fastq output.
*  summary.csv file
]

.pull-right[
BAM TAGS .small[useful]
*  BX - corrected barcode
*  QX - barcode quality

.tiny[`samtools view $bamfile`]  
.tiny[`&lt;header&gt; &lt;flags&gt; &lt;seq&gt; &lt;seqQ&gt; BX:Z:AAACACCAGGGTTCGA`]
]

???

---

# Longranger align
BWA + 10X Lariat aligner
.pull-left[
Input: 
 + Fastq.&lt;br&gt;
~Run time: 
 + 24 hours.&lt;br&gt;
Output: 
 + Aligned BAM
 + Alignment QC stats
]
.pull-right[
Output bam tags:
.baby-bear[
* AS - Primary Alignment Score
* XS - Secondary Alignment Score
* TBC.
]
]

???

Takes about a day, what bam tag outputs are provided.
First pass alignment with BWA.
Second pass with the 10X lariat aligner.
---

# Lariat Aligner

Extra benefits
Uses primary and secondary alignments alongside information of the GEM tag to:
1. Boost confidence in the alignment score (AM)
2. Use alignments with the same molecular tag to align to  repeat regions (XS)
3. Overcome tandem duplications using GEM tag of secondary alignments (XT)

???


---

# Longranger wgs
Obtain SNP calling .small[GATK], phasing and SV information

Input: 
 + Fastq.
~ Run time: 
 + Three days .small[w. 250GB RAM and 16 cores]
Output:
 + List of structural variants in bed format.
 + BAM files with Phasing tags.
 + .loupe file.

???


---

# The Loupe file
~ 5Gb file.

Contains alignment information across the entire genome.
Similar to an IGV browser.

&lt;img src=./images/full_sv_view.png width=60%&gt;

???


---

# Viewing your loupe-file

Live Demonstration on a HPC.

Steps:
1. Allocate a node (be greedy)
2. Copy all your .loupe files to a local space on the HPC node.
3. Start loupe.
4. ssh in (with binding) into that node.

All code in the comments

???

Homozygous deletion:
chr3:129,663,385-129,863,385

Heterozygous deletion:
chr1:152,355,549-152,755,549
check the linked reads tab as well for this one.

Duplication:
chr6:165,000-495,000

Het DUP:

Homo INV:
32: 7 54,287,991, 54376,000

`ssh username@headnode`
`screen`
`srun --mem=100G --time=2-0 --pty bash`
`export LOUPE_SERVER=/path/to/loupe files`
`start-loupe.sh`

`ssh -L3000:127.0.0.1:3000 username@workernode`

---

# List of resources used today

.pull-left[
Miscellaneous:
*  [ssh-bind ports](https://help.ubuntu.com/community/SSH/OpenSSH/PortForwarding)
*  [screen]
*  [slurm]
]

.pull-right[
10X video
10X BAM Tags
Loupe downloader
NA12878 10X download
]

---

# Thank yous

???

todo thank yous.
    </textarea>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "pygments",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function() {
  var d = document, s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})();</script>

<script>
(function() {
  var i, text, code, codes = document.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
})();
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
