<!DOCTYPE html>
<html>
  <head>
    <title>Longranger, Lariat and Loupe</title>
    <meta charset="utf-8">
    <meta name="author" content="Alexis Lucattini" />
    <link rel="stylesheet" href="libs\remark-css-0.0.1\linked_reads.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# Longranger, Lariat and Loupe
## The ins and outs of the 10X pipeline
### Alexis Lucattini
### Fri 27 Jul 2018

---


### Another visual introduction to linked-reads:

.small[Because you can never watch this video too many times]
.center[&lt;a href="https://10xgenomics.wistia.com/medias/2r0nal3n3j"&gt;&lt;img src=./images/intro_video_thumbnail.png width=68%&gt;]

???

Things to note:

The GEMs, 14 bp barcodes, and molecular tags.

Add linked reads video here.
https://10xgenomics.wistia.com/medias/2r0nal3n3j

70s in.

---

# A Micro-Glossary

* Barcode
  + .small[A unique set of 14 nucleotides.]
* Molecule:
  + .small[The ~50Kb input molecule into the 10X system.]
* Read:
  + .small[~200 bp containing genomic information and a barcode.]
* GEM:
  + .small[A single droplet containing identical barcodes.]
  + .small[Approximately 10 molecules exist inside per GEM.]

---

# The Longranger pipeline.

.center[&lt;img src=./images/10X_DIAGRAM.png width=50%&gt;]

???

An important note is that the bcl2fastq stage is separate from the standard illumina bcl2fastq.
The longranger mkfastq is essentially a wrapper aroudn bcl2fastq with a couple of tinkerings.

You then have the option of 'basic, align or wgs from fastq', each with cumulative benefits.
You may wish to use these two on the left for running alternative analysis callers not menitoned today. 

---

## 10x mkfastq
.pull-left[
.baby-bear[
* Wrapper of bcl2fastq
* Input
  + Requires input and output directories
  + A samplesheet
  + A base-mask parameter.
* Output
  + Fastq files split by 10X kit ID .tiny[(SI-GA-01)]
    + Barcode/Index QC Values.
    + GEM estimates
]]
.pull-right[
*  `*_L1.fastq.gz` contains the i7 index
*  `*_R1.fastq.gz` contains the fastq information. &lt;br&gt;&lt;br&gt;
&lt;img src=./images/index_orientation.png width=130%&gt;
]

???



---

# BAM Files
```
samtools view align/MG10_output/outs/possorted_bam.bam | head -n1
```  
.small[
```
A00121:44:H5CGNDSXX:3:1636:0:1306618    177     chr1    9990    
20      13S9M1D42M6I24M6I18M6I26M    ATCTCTATCATTATGTCTGTCCATAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCC 
F,F,:,F,,F,,F:,:,F,FFF,FFFFFF:FFFFFFF::FFFFFFFFFFFFFFFF:FFF:FFFFF  
RX:Z:CAAGATCTCCGACGAC   QX:Z:FFFFFFFFFFFFFFFF   BC:Z:GTGGTACC   QT:Z:FFFFFFFF   XS:i:-64        AS:i:-62        XM:Z:0  AM:Z:0  XT:i:0  BX:Z:CAAGATCTCCGACGAC-1  OM:i:20
```
]

---

# Longranger basic
.pull-left[
* Input: 
  + Fastq
* Run time: 
  + A few hours.&lt;br&gt;&lt;br&gt;
* Output:
  + unaligned Bam or fastq output.
  + summary.csv file
]
.pull-right[
* BAM TAGS 
  + BX - corrected barcode
  + QX - barcode quality
]

???

---

# Longranger align
BWA + 10X Lariat aligner
.pull-left[
.baby-bear[
* Input:
  + Fastq.&lt;br&gt;&lt;br&gt;
* Run time:
 + 24 hours.&lt;br&gt;&lt;br&gt;
* Output:
 + Aligned BAM
 + Alignment QC stats
]
]
.pull-right[
Output bam tags:
.baby-bear[
* AS - Primary Alignment Score
* XS - Secondary Alignment Score
* To be cont..
]
]

???

Takes about a day, what bam tag outputs are provided.
First pass alignment with BWA.
Second pass with the 10X lariat aligner.
---

# Lariat Aligner

Extra alignment help, keeps both primary and secondary alignments alongside information of the GEM tag to:
1. Boost confidence in the alignment score (AM)
2. Use alignments with the same molecular tag to align to  repeat regions (XS)
3. Overcome tandem duplications using GEM tag of secondary alignments (XT)

???

AM tag: Increase the alignment score if the alignment is in a long molecule.
XS: Did the aligner need assistance in flanking regions to align
XT: Mark tandem duplications.

---

# Longranger wgs
Obtain SNP calling .small[(GATK)], phasing and SV information

* Input: 
 + Fastq.
* ~ Run time: 
 + Three days .small[w. 250GB RAM and 16 cores]
* Output:
 + .small[List of structural variants in .bed format.]
 + .small[BAM files with additional haplotype phasing tags.]
 + .small[.loupe file.]

???


---

# The Loupe file

.pull-left[
~ 5Gb file.  
Contains alignment information across the entire genome.

]
.pull-right[
&lt;img src=./images/full_sv_view.png width=110%&gt;
]

???

Similar to an IGV

---

# Viewing your loupe-file

Live Demonstration on a HPC.

Steps:
1. Allocate a node (be greedy)
2. Copy all your .loupe files to a local space on the HPC node.
3. Start the loupe application.
4. ssh in (with binding) into that node.

All code in the comments (press p)

???

Homozygous deletion:
chr3:129,663,385-129,863,385

Heterozygous deletion:
chr1:152,355,549-152,755,549
check the linked reads tab as well for this one.

Duplication:
chr16:34,080,000-34,880,000;chr16:34,230,000-35,030,000
chr6:0-626,484;chr6:0-626,484

Homo INV:
chr12:17,723,137-18,123,137;chr12:17,813,072-18,213,072
`ssh username@headnode`
`screen`
`srun --mem=100G --time=2-0 --pty bash`
`export LOUPE_SERVER=/path/to/loupe files`
`start-loupe.sh`

`ssh -L3000:127.0.0.1:3000 username@workernode`

---

# List of resources used today

.pull-left[
Miscellaneous:
*  [ssh-bind ports](https://help.ubuntu.com/community/SSH/OpenSSH/PortForwarding)
*  [screen](https://www.rackaid.com/blog/linux-screen-tutorial-and-how-to)
*  [slurm](https://slurm.schedmd.com/quickstart.html)
]

.pull-right[
10X related resources
[10X video](https://10xgenomics.wistia.com/medias/2r0nal3n3j)
[10X BAM Tags](https://support.10xgenomics.com/genome-exome/software/pipelines/latest/output/bam)
[Loupe downloader](https://support.10xgenomics.com/genome-exome/software/downloads/latest)
[NA12878 10X download](https://support.10xgenomics.com/genome-exome/datasets/2.2.1/NA12878_WGS_v2)
]

---

## Thank yous

.pull-left[
AGRF
.baby-bear[
* R&amp;D:
    + Jafar Jabbari
    + Kirby Siemering
    + Azadeh Seidi (GIH)
* Lab Ops:
    + Matthew Tinning
* Bioinformatics:
    + Everyone!
* IT:
    + Douglas Morrison
]
]
.pull-right[
MGHA
.baby-bear[
*  New Tech. Working Group
  + Natalie Thorne
  + Stan Baudinette
  + Simon Sadedin
  + Sebastian Lunke
  + Melanie O'Keefe
  + Sarah Payton
  + Juny Kesumadewi
]
]

???

todo thank yous.
    </textarea>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "pygments",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function() {
  var d = document, s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})();</script>

<script>
(function() {
  var i, text, code, codes = document.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
})();
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
