<!DOCTYPE html>
<html>
  <head>
    <title>Nanopore and Minimap2 things</title>
    <meta charset="utf-8">
    <meta name="author" content="Alexis Lucattini" />
    <link rel="stylesheet" href="libs\remark-css-0.0.1\nanopore_dunedin.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# Nanopore and Minimap2 things
## <br> PeterMac Floor 12
### Alexis Lucattini
### Thu 30th May 2018

---

# Nanopore Overview
.pull-left[
.baby-bear[
## Pros:
Ultra-long reads.
Low-cost.
Large Community support
Quick library prep
]
]
.pull-right[
.baby-bear[
## Cons:
Low accuracy.
Low yield.
Systematic bias - homopolymers.
Sample Specific.
]
]

---

# The PromethION!

Kit-9, sample sensitive, great N50!

.pull-left[
&lt;img src=./images/prom.yield.png width=90%&gt;
]
.pull-right[
&lt;img src=./images/prom.hist.png width=90%&gt;
]

---

# Alignment algorithms
Initially limited, due to unique sequencing error parameters.

.baby-bear[
* LAST was the first best choice.
  + Still very useful for metagenomics.&lt;br&gt;&lt;br&gt; 
* Bwa mem with optimised parameters.  
  + -ont2d option equivalent to "-k14 -W20 -r10 -A1 -B1 -O1 -E1 -L0"&lt;br&gt;&lt;br&gt; 
* Minimap2 with the map-ont option. 
  + Equivalent to "-k15"
]

---

# Flags and Tags
.small[
Each alignment have set flags and tags. 
This allows you to group reads/alignments based on certain conditions.  
* Flags provide a boolean value for a given condition - True/False.
  + *range: 0-255.*
  + *Boolean combinations for eight attributes.*&lt;br&gt;&lt;br&gt; 
* Tags can be optional and hold any type of information.
  + *int, boolean, string*&lt;br&gt;&lt;br&gt; 
* Many flags/tags are illumina-sequencing specific:
  + *paired end*
]
---

## Important tags to look for in nanopore sequencing
.pull-left[
.small[
* NM tags
  + The edit distance between read and sequence.
  + 1 - NM/AlignmentLength ~= Accuracy
* MD tag 
  + Verbose version of the cg tag (CIGAR string) of an alignment
  + Provides SNP/indel information
  + No clip info
  + Insertions are not recorded. Info  [here](https://github.com/vsbuffalo/devnotes/wiki/The-MD-Tag-in-BAM-Files)
]
]

.pull-right[

&lt;table class="table" style="font-size: 20px; margin-left: auto; margin-right: auto;"&gt;
&lt;tbody&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;font-family: monospace;"&gt; REF &lt;/td&gt;
   &lt;td style="text-align:left;font-family: monospace;"&gt; TAA-GCAAGGTCG &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;font-family: monospace;"&gt;  &lt;/td&gt;
   &lt;td style="text-align:left;font-family: monospace;"&gt; ||||||||||||| &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;font-family: monospace;"&gt; READ &lt;/td&gt;
   &lt;td style="text-align:left;font-family: monospace;"&gt; TATGGCA--GTCG &lt;/td&gt;
  &lt;/tr&gt;
&lt;/tbody&gt;
&lt;/table&gt;
.small[
* A CIGAR string
  + `3M1I3M2D2M4`  
* An MD string equivalent
  + `2MA3^AG4`  
* A CS tag (See next page)
  + `:2*AT+G:3-AG:4`
]
]

---

# The CS tag and MD tag
.pull-left[
.baby-bear[
**CS tag**
* Verbose version of MD tag
* Easier for parsing.
* Relatively new.
* MD tag still required for some parsers.
  + Pipe through  `samtools calmd` prior to sorting.

CS tag example:  
`:6-ata:10+gt:4*at:3`
]
]
.pull-right[
.baby-bear[
Key:
* `:[0-9]+ is an identical block`
* '-[ACTG]+ is a deletion`
* '+[ACTG]+ is an insersion
* '*XY' is a substitution with base X subsituted for base y.
]
]

---

# The --cs-long option

The `--cs-long` parameter provides a reference free solution to alignment queries.

.pull-left[

&lt;table class="table" style="font-size: 20px; margin-left: auto; margin-right: auto;"&gt;
&lt;tbody&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;font-family: monospace;"&gt; REF &lt;/td&gt;
   &lt;td style="text-align:left;font-family: monospace;"&gt; TAA-GCAAGGTCG &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;font-family: monospace;"&gt;  &lt;/td&gt;
   &lt;td style="text-align:left;font-family: monospace;"&gt; ||||||||||||| &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;font-family: monospace;"&gt; READ &lt;/td&gt;
   &lt;td style="text-align:left;font-family: monospace;"&gt; TATGGCA--GTCG &lt;/td&gt;
  &lt;/tr&gt;
&lt;/tbody&gt;
&lt;/table&gt;
.small[
CS long tag example:
  + `=TA*AT+G=GCA-AG=GTCG`
]
]
.pull-right[
.baby-bear[
**Key**
+ `=[ATCG]+` is a set of matches
+ `-[ACTG]+` is a deletion
+ `+[ACTG]+` is an insertion
+ *XY' is a substitution with X-ref for Y-read.
]
]

---

# For the love of regex

.large[`/(:[0-9]+|\*[a-z][a-z]|[=\+\-][A-Za-z]+)+/`]

---

# Alignments in Minimap2
.pull-left[
.baby-bear[
**Supplementary alignments**:
Definition: 
* A segment of the read which is (mostly) non-overlapping the primary alignment that maps to an alternative position in the genome.

Cases:
* Structural variation.
]
]
.pull-right[
.baby-bear[
**Secondary alignments**:
Definition: 
* A segment of the read which provides an alternative alignment to the primary.

Cases:
* Repeat regions of the genome.
]
]

---

# Pysam, pandas and puerility

.baby-bear[
1. Pysam to pull out the cs string and other metadata from an alignment segment.
2. Convert the alignment segment to a pandas dataframe with the following columns:
+ ReadNuc, RefNuc  `# Nucleotide`
+ ReadPos, RefPos  `# Position in read/ref`
+ ReadStartContig, RefStartContig  `# Start of indel/match`
+ ReadEndContig, RefEndContig  `# End of indel/match`
]
---

# Thank yous.
.pull-left[
.baby-bear[
* R&amp;D:
    + Azadeh Saedi
    + Ken McGrath
    + Kirby Siemering
    + Lavinia Gordon
* Lab Ops:
    + Matthew Tinning
* IT:
    + Chris Hunt
    + Douglas Morrison
    + Gismon Thomas
* PR:
    + Desley Pitcher
]
]
.pull-right[
.baby-bear[
* WEHI
    + Matthew Ritchie
    + Quentin Gouil
    + Scott Gigante
    + Chris Woodruff
]
]
    </textarea>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "pygments",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function() {
  var d = document, s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})();</script>

<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre']
  }
});
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
